diff --git a/.clippy.toml b/.clippy.toml
index 1d4661b6cb..0a5dc219be 100644
--- a/.clippy.toml
+++ b/.clippy.toml
@@ -1,4 +1,8 @@
+# big constructors
 too-many-arguments-threshold = 20
+# we want to fix the enum suffixes, so we need to disable that
+avoid-breaking-exported-api = false
+
 disallowed-methods = [
   # we use tracing with the log feature instead of the log crate.
   { path = "log::info", reason = "use tracing::info instead" },
diff --git a/consensus/.clippy.toml b/consensus/.clippy.toml
index 575b80b1c1..701e9d866f 100644
--- a/consensus/.clippy.toml
+++ b/consensus/.clippy.toml
@@ -4,6 +4,8 @@ cognitive-complexity-threshold = 100
 type-complexity-threshold = 10000
 # big constructors
 too-many-arguments-threshold = 20
+# we want to fix the enum suffixes, so we need to disable that
+avoid-breaking-exported-api = false
 
 disallowed-methods = [
   { path = "anyhow", reason = "we prefer to use eyre" },
diff --git a/narwhal/.clippy.toml b/narwhal/.clippy.toml
index 2ca07227b4..a27c9dfa1b 100644
--- a/narwhal/.clippy.toml
+++ b/narwhal/.clippy.toml
@@ -4,6 +4,8 @@ cognitive-complexity-threshold = 100
 type-complexity-threshold = 10000
 # big constructors
 too-many-arguments-threshold = 20
+# we want to fix the enum suffixes, so we need to disable that
+avoid-breaking-exported-api = false
 
 disallowed-methods = [
   { path = "anyhow", reason = "we prefer to use eyre" },
